### ISSUE TITLE
Minor improvement in AP during training

### ISSUE THREAD

**Issue by WUyinwei-hah** [frank-xwang/InstanceDiffusion#32]
Hi, very great work!
I'm trying to reproducing your training result.
Now I'm training with batchsize of 124 for 2W steps. But I found the AP on COCO is very low, as shown in figure below:
![image](https://github.com/frank-xwang/InstanceDiffusion/assets/56622077/d48ef340-47af-414b-8d44-e6be83e6b16b)
The AP is only 0.041, so I wonder is it normal that the metrics improve slowly during the training?
If not, could you provide me with any suggestions toward training?
Thanks!

**Comment by WUyinwei-hah** [comment]
I further checked the samples generated from function _save_ckpt_and_result_, and found it looks pretty good.
So is the above phenomenon caused by the poor generalization from LAION to COCO?

**Comment by frank-xwang** [comment]
Hi, apologies for the very delayed response :-(. Hopefully, this is still helpful to you or perhaps useful for others as well.

We actually didn’t use COCO samples in our training process, so I don’t think the issue is primarily due to poor generalization from LAION to COCO. However, there’s one aspect you might want to consider: I prioritized selecting images with more bounding boxes per image as a key criterion during training. Additionally, you could enhance your training set by incorporating images (along with their ground-truth bounding box labels) from datasets like Visual Genome, Object365, and CC3M. This could help improve performance, particularly in terms of box-based position control. 

I hope this helps! Let me know if you have further questions. 😊
